libraries/lib-graphics/graphics/Size.h

```diff
@@ -0,0 +1,175 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  Size.h
+
+  Dmitry Vedenko
+
+**********************************************************************/
+#pragma once
+
+#include <cmath>
+#include <type_traits>
+
+namespace graphics
+{
+namespace details
+{
+template <typename DataType>
+constexpr DataType GetPositiveSizeValue(DataType value) noexcept
+{
+   if constexpr (std::is_unsigned_v<DataType>)
+      return value;
+   else
+      return value >= DataType {} ? value : -value;
+}
+} // namespace details
+
+//! A class that represents size in 2D space.
+template <typename DataType>
+struct SizeType final
```

Comment:
Maybe `Size` could turn into the 2d vector type, distinguished from a point.  A point and a diagonal vector determine a rectangle.